zerver/lib/actions.py

```diff
@@ -1819,14 +1819,19 @@ def do_send_messages(
             # Service bots (outgoing webhook bots and embedded bots) don't store UserMessage rows;
             # they will be processed later.
             mentioned_user_ids = send_request.message.mentions_user_ids
+
+            # Extend the list with users who have muted the sender.
+            mark_as_read_msg = list(mark_as_read)
+            mark_as_read_msg.extend(get_muting_users(send_request.message.sender))
```

Comment:
I think it'd be worth using `Set` here; e.g. `get_muting_users` could return a set, and then you could just add everything in the `mark_as_read` parameter to it (avoiding mutating that parameter).

Reply:
This now uses a `Set`, just like other inputs to `create_user_messages`.